### PR TITLE
filename pattern

### PR DESCRIPTION
**Pull request by dsavchenko**

### CODE REVIEW

File: nb2workflow/nbadapter.py [hunk 1]
```diff
@@ -557,17 +557,22 @@ def remove_tmpdir(self):
 def notebook_short_name(ipynb_fn):
     return os.path.basename(ipynb_fn).replace(".ipynb","")
 
-def find_notebooks(source, tests=False) -> Dict[str, NotebookAdapter]:
+def find_notebooks(source, tests=False, pattern = r'.*') -> Dict[str, NotebookAdapter]:
 
-    base_filter = lambda fn: "output" not in fn and "preproc" not in fn
+    def base_filter(fn): 
+        good = "output" not in fn and "preproc" not in fn
+        if pattern != r'.*':
```
Comment (volodymyrss) [inline nb2workflow/nbadapter.py:564]: this condition is redundant since this regex always matches, right?

File: nb2workflow/service.py [hunk 2]
```diff
@@ -693,7 +694,7 @@ def main():
         root.setLevel(logging.INFO)
         handler.setLevel(logging.INFO)
 
-    app.notebook_adapters = find_notebooks(args.notebook)
+    app.notebook_adapters = find_notebooks(args.notebook, pattern = args.pattern)
```
Comment (volodymyrss) [inline nb2workflow/service.py:697]: let's have `pattern=args..` - no spaces around "="

**Comment by volodymyrss** [comment]
we have tests around this function, could you add one or two?

**Comment by dsavchenko** [comment]
> we have tests around this function, could you add one or two?

ok

**Comment by dsavchenko** [comment]
and I will also add the `pattern` option in `deploy.py`

**Review by volodymyrss — APPROVED** [review]

**Pull request merged**